src/entity/SquadPostModeration.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  ManyToOne,
+  PrimaryColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import { Post } from './posts';
+import { Source } from './Source';
+import { User } from './user';
+
+export enum SquadPostModerationStatus {
+  Approved = 'approved',
+  Rejected = 'rejected',
+  Pending = 'pending',
+}
+
+@Entity()
+export class SquadPostModeration {
+  @PrimaryColumn({ type: 'text' })
+  id: string;
+
+  @Column({ type: 'text' })
+  sourceId: string;
+
+  @ManyToOne('Source', (source: Source) => source.id, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  source: Promise<Source>;
```

Comment:
Do we need both `sourceId` and `source`? I think that We can go for the relation only (same for post and sharedPost).
cc @capJavert

Reply:
Yes, that is true for the schema. However, for the entity we need the FK which is the source id.

Reply:
Yeah, source is not in the db, its just a logical relation on a class.